File: Assignment-2/sec-2.py
# Closure example
def make_adder(x):
    def add(y):
        return x + y   # x is captured from outer scope
    return add

add10 = make_adder(10)
print(add10(5))   # Output: 15

# Dynamic typing example
x = "10"
print(x + "5")    # Output: 105

File: Assignment-2/sec-2.js
// Closure example
function makeAdder(x) {
    return function (y) {
        return x + y;   // x is captured from outer scope
    };
}

const add10 = makeAdder(10);
console.log(add10(5));   // Output: 15

// Dynamic typing + implicit coercion
console.log("10" + 5);   // Output: "105"
console.log("10" - 5);   // Output: 5
